launcher/launcher/main.go

```diff
@@ -180,3 +190,59 @@ func startLauncher() error {
 
 	return r.Run(ctx)
 }
+
+// verifyFsAndMount checks the partitions/mounts are as expected, based on the command output reported by OS.
+func verifyFsAndMount() error {
+	// check protected_stateful_partition is encrypted and is on integrity protection
+	cryptsetupOutput, err := exec.Command("cryptsetup", "status", "/dev/mapper/protected_stateful_partition").Output()
+	if err != nil {
+		return err
+	}
+	matched := regexp.MustCompile(`type:\s+LUKS2`).FindString(string(cryptsetupOutput))
+	if len(matched) == 0 {
+		return fmt.Errorf("stateful partition is not LUKS2 formatted: \n%s", cryptsetupOutput)
+	}
+	matched = regexp.MustCompile(`integrity:\s+aead`).FindString(string(cryptsetupOutput))
+	if len(matched) == 0 {
+		return fmt.Errorf("stateful partition is not integrity protected: \n%s", cryptsetupOutput)
+	}
+
+	// make sure /var/lib/containerd is on protected_stateful_partition
+	findmountOutput, err := exec.Command("findmnt", "/dev/mapper/protected_stateful_partition").Output()
+	if err != nil {
+		return err
+	}
+	matched = regexp.MustCompile(`/var/lib/containerd\s+/dev/mapper/protected_stateful_partition\[/var/lib/containerd\]`).FindString(string(findmountOutput))
+	if len(matched) == 0 {
+		return fmt.Errorf("/var/lib/containerd was not mounted on the protected_stateful_partition: \n%s", findmountOutput)
+	}
+
+	// check /tmp is on tmpfs
+	findmntOutput, err := exec.Command("findmnt", "tmpfs").Output()
+	if err != nil {
+		return err
+	}
+	matched = regexp.MustCompile(`/tmp\s+tmpfs\s+tmpfs`).FindString(string(findmntOutput))
+	if len(matched) == 0 {
+		return fmt.Errorf("/tmp was not mounted on the tmpfs: \n%s", findmntOutput)
+	}
+
+	// check verity status on vroot and oemroot
+	// TODO @jiankun Verify the content using the extracted hashes from cmdline
```

Comment:
some command like `dmsetup table vroot`. Since this is just a simple check against misconfiguration, I'm not sure if checking hashes is worth it, but go for it

Reply:
make sense, will leave it for now